source/parser/independent/intermediate-instruction.cpp

```diff
@@ -0,0 +1,259 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string>
+#include <vector>
+
+#include "arch/common/architecture.hpp"
+#include "arch/common/architecture.hpp"
+#include "common/assert.hpp"
+#include "common/utility.hpp"
+#include "core/conversions.hpp"
+#include "core/memory-access.hpp"
+#include "parser/common/code-position-interval.hpp"
+#include "parser/common/code-position.hpp"
+#include "parser/common/compile-error-list.hpp"
+#include "parser/common/final-representation.hpp"
+#include "parser/independent/enhance-symbol-table-immutable-arguments.hpp"
+#include "parser/independent/execute-immutable-arguments.hpp"
+#include "parser/independent/intermediate-instruction.hpp"
+#include "parser/independent/macro-directive.hpp"
+#include "parser/independent/positioned-string.hpp"
+#include "parser/independent/preprocessing-immutable-arguments.hpp"
+#include "parser/independent/section-tracker.hpp"
+#include "parser/independent/symbol-graph.hpp"
+#include "parser/independent/symbol-replacer.hpp"
+#include "parser/independent/syntax-tree-generator.hpp"
+
+using size_t = std::size_t;
+
+IntermediateInstruction::IntermediateInstruction(
+    const CodePositionInterval& positionInterval,
+    const std::vector<PositionedString>& labels,
+    const PositionedString& name,
+    const std::vector<PositionedString>& sources,
+    const std::vector<PositionedString>& targets)
+: IntermediateOperation(positionInterval, labels, name)
+, _sources(sources)
+, _targets(targets) {
+}
+
+void IntermediateInstruction::execute(
+    const ExecuteImmutableArguments& immutable,
+    CompileErrorList& errors,
+    FinalCommandVector& commandOutput,
+    MemoryAccess& memoryAccess) {
+  // For a machine instruction, it is easy to "execute" it: just insert it
+  // into the final form.
+  auto compiled = compileInstruction(immutable, errors, memoryAccess);
+  if (compiled.node()) {
+    commandOutput.push_back(compiled);
+  }
+}
+
+std::vector<std::shared_ptr<AbstractSyntaxTreeNode>>
+IntermediateInstruction::compileArgumentVector(
+    const std::vector<PositionedString>& vector,
+    const ExecuteImmutableArguments& immutable,
+    CompileErrorList& errors,
+    MemoryAccess& memoryAccess) {
+  std::vector<std::shared_ptr<AbstractSyntaxTreeNode>> output;
+
+  SymbolReplacer::DynamicReplacer labelReplacerFunction = [&](
```

Comment:
It's better to explicitly capture the variables you need. For example, you're capturing the `vector` here, but not using it. So `[&immutable]` will do